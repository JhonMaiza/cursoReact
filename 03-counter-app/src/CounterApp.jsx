import Prototype from  'prop-types';
import { useState } from 'react';
export const CounterApp = ( { value } ) => {

  const [ counter, setCounter ] = useState(value)

  const handleAdd = () => {
    setCounter( counter + 1 );
    //setCounter( c => c + 1 )
  }
  const handleSubtract = () => {
    setCounter( counter - 1 );
  }

  const handleRestart = () => {
    setCounter( value );
  }
  return (
    <>
      <h1> Counter App </h1>
      <h2>{ counter }</h2>
      <button onClick={handleAdd}>
        Aumentar
      </button>

      <button onClick={handleSubtract}>
        Restar
      </button>

      <button aria-label='btn-reset' onClick={handleRestart}>
        Restablecer
      </button>
    </>
  )
}

CounterApp.propTypes = {
  value: Prototype.number.isRequired,
}

// CounterApp.defaultProps = {
//   value: 12,
// }
