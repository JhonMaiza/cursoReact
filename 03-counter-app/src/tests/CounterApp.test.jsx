import { fireEvent, render, screen } from "@testing-library/react";
import {CounterApp} from '../CounterApp'

describe('Prueba CounterApp', () => {
  const value = 10;
  test('debe hacer match con el snapshot', () => {
    const { container } = render(<CounterApp value={1}/>)
    expect( container ).toMatchSnapshot();
  });

  test('debe mostar el valor inicial de 100 ', () => {
    render(<CounterApp value={100}/>);
    expect( screen.getByText( 100 )).toBeTruthy();
  });

  test('debe de incrementar con el boton +1', () => {
    render(<CounterApp value={100}/>);
    fireEvent.click( screen.getByText('Aumentar') );
    expect( screen.getAllByText('101')).toBeTruthy();
  });

  test('debe de decrementar con el boton +1', () => {
    render(<CounterApp value={100}/>);
    fireEvent.click( screen.getByText('Restar') );
    expect( screen.getAllByText('99')).toBeTruthy();
    
  });
  
  test('debe de funcionar de reset', () => {
    render(<CounterApp value={355}/>);
    fireEvent.click( screen.getByText('Aumentar') );
    fireEvent.click( screen.getByText('Aumentar') );
    fireEvent.click( screen.getByText('Aumentar') );
    //fireEvent.click( screen.getByText('Restablecer') );
    fireEvent.click( screen.getByRole('button', { name: 'btn-reset'}));
    expect( screen.getByText('355')).toBeTruthy();
  });
})
