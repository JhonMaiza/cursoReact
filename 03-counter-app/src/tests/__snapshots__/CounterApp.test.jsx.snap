// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba CounterApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
     Counter App 
  </h1>
  <h2>
    1
  </h2>
  <button>
    Aumentar
  </button>
  <button>
    Restar
  </button>
  <button
    aria-label="btn-reset"
  >
    Restablecer
  </button>
</div>
`;
